Minor fixes to payment channels commands (RIPD-1466 and RIPD-1467)

seelabs opened the pull request.

seelabs commented on src/ripple/rpc/handlers/PayChanClaim.cpp:
```diff
@@ -90,15 +90,31 @@ Json::Value doChannelVerify (RPC::Context& context)
     auto const& params (context.params);
     for (auto const& p :
-        {jss::public_key, jss::channel_id, jss::amount, jss::signature})
+                 {jss::public_key, jss::channel_id, jss::amount, jss::signature})
```
Fixed

seelabs commented:
Rebased onto 0.80.0-b4

miguelportilla commented on src/ripple/rpc/handlers/PayChanClaim.cpp:
```diff
@@ -90,15 +90,31 @@ Json::Value doChannelVerify (RPC::Context& context)
+            pk.emplace(makeSlice(pkHex.first));
+        }
+
+        if (!pk)
```
This `if` should be moved to after the `emplace.`

seelabs replied:
It's not needed. Fixed.

miguelportilla commented on src/ripple/net/impl/RPCCall.cpp:
```diff
@@ -669,7 +670,21 @@ class RPCParser
+
+            std::pair<Blob, bool> pkHex(strUnHex (strPk));
+            if (!pkHex.second)
+                return false;
```
Suggestion: This code path is not covered by unit test. We can improve the coverage and test by testing with a non-hexidecimal key.

miguelportilla commented on src/ripple/net/impl/RPCCall.cpp:
```diff
@@ -669,7 +670,21 @@ class RPCParser
+                return false;
+
+            if (!publicKeyType(makeSlice(pkHex.first)))
+                return false;
```
Suggestion: This code path is not covered by unit test. We can improve the coverage and test by testing with hexidecimal key of an invalid length.

miguelportilla commented on src/ripple/net/impl/RPCCall.cpp:
```diff
@@ -669,7 +670,21 @@ class RPCParser
+            if (!publicKeyType(makeSlice(pkHex.first)))
+                return false;
+
+            return true;
```
Same here

miguelportilla commented on src/ripple/rpc/handlers/PayChanClaim.cpp:
```diff
@@ -90,15 +90,31 @@ Json::Value doChannelVerify (RPC::Context& context)
+        {
+            std::pair<Blob, bool> pkHex(strUnHex (strPk));
+            if (!pkHex.second)
+                return rpcError(rpcPUBLIC_MALFORMED);
```
Suggestion: This code path is not covered by unit test. We can improve the coverage and test.

miguelportilla commented on src/ripple/rpc/handlers/PayChanClaim.cpp:
```diff
@@ -90,15 +90,31 @@ Json::Value doChannelVerify (RPC::Context& context)
+                return rpcError(rpcPUBLIC_MALFORMED);
+            auto const pkType = publicKeyType(makeSlice(pkHex.first));
+            if (!pkType)
+                return rpcError(rpcPUBLIC_MALFORMED);
```
Suggestion: This code path is not covered by unit test. We can improve the coverage and test.

seelabs commented:
Added a commit to fix an issue with `std::stoul` and add tests. Also rebased to 0.90.0.

@miguelportilla The places you tagged in an earlier appear to be tested even before this commit. I'm guessing github put the comment on the wrong line of code (I've seen it do this in other commits). I'm assuming you meant to tag the untested code that the new commit now tests.

ripplelabs-jenkins commented:
## Jenkins Build Summary

Built from [this commit](https://github.com/ripple/rippled/commit/825ab8cdbf9d30615425dad22751c9c8787c5722)

Built at __20171208 - 21:10:22__

### Test Results

Build Type | Result | Status
---------- | ------ | ------
clang.debug.unity | 970 cases, 0 failed, t: 375s | PASS :white_check_mark: 
coverage | 970 cases, 0 failed, t: 588s | PASS :white_check_mark: 
clang.debug.nounity | 968 cases, 0 failed, t: 319s | PASS :white_check_mark: 
gcc.debug.unity | 970 cases, 0 failed, t: 419s | PASS :white_check_mark: 
gcc.debug.nounity | 968 cases, 0 failed, t: 324s | PASS :white_check_mark: 
clang.release.unity | 969 cases, 0 failed, t: 438s | PASS :white_check_mark: 
gcc.release.unity | 969 cases, 0 failed, t: 460s | PASS :white_check_mark:

scottschurr commented on src/ripple/rpc/handlers/PayChanClaim.cpp:
```diff
@@ -31,8 +31,37 @@
 namespace ripple {
 
+static
+boost::optional<std::uint64_t>
```
Since there are now two identical implementations of this function, how would you feel about putting it in `basics/StringUtilities`? That way if there are future bugs to fix they will only need to be fixed in one place.

seelabs replied:
I considered that, but I decided they shouldn't be project-wide utilities. I agree code duplication like this is bad, but too many project-wide utility functions has a cost too. It makes finding the right utility function hard to find because the useful ones are mixed in with ones like this that are only used in two files.

If the two places were in the same subsystem, I'd look for a place in that subsystem to put them. As is, they are in `ripple/net` and `ripple/rpc`, so the util would have to live in `ripple`. 

I'm OK promoting this project-wide if you feel strongly, but I still vote for the two (small) duplicated functions.

scottschurr replied:
I don't feel strongly about it.  I have a lot of sympathy for the costs associated with too many project-wide utility functions.  I regularly thrash about trying to find where some utility function is hidden.  I also probably accidentally create duplicate code, since I have no idea that a function already exists that does what I want to do.  I'm not sure precisely what to do about either of those problems.

At any rate, my personal take is that this is not an overly specialized function, so I'd be fine with promoting it as a general facility (since it took you a couple of tries to get it to its current state of correctness).  But I'm also okay with duplicating it.

seelabs commented:
Squashed and rebased onto 0.90.0-b2

seelabs commented:
In 0.90.0-b3
